fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/client/item/v1/AnimationUpdateHandler.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.item.v1;
+
+import net.minecraft.item.ItemStack;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+
+/**
+ * Provides information on whether an {@link net.minecraft.item.Item} should run its update animation, such as for NBT updates.
+ */
+@Environment(EnvType.CLIENT)
+public interface AnimationUpdateHandler {
+	/**
+	 * Called once per tick to check whether the given held {@link ItemStack} should run the update animation.
+	 *
+	 * <p>In vanilla, the update animation is run if an {@link ItemStack} is not {@linkplain ItemStack#areEqual
+	 * equal} to its previous version, such as any NBT value change.
+	 * The {@code updated} is the most up-to-date version, while the {@code original} is a cached version from the previous tick.
+	 * Both {@link ItemStack} instances are guaranteed to have the same {@linkplain ItemStack#getItem() underlying item}.
+	 *
+	 * @param original cached {@link ItemStack} from previous tick
+	 * @param updated updated {@link ItemStack} to check for updates on
+	 * @return whether or not the item update animation should play
+	 */
+	@Environment(EnvType.CLIENT)
+	boolean updateAnimation(ItemStack original, ItemStack updated);
```

Comment:
```suggestion
	boolean allowUpdateAnimation(ItemStack original, ItemStack updated);
```

Reply:
should start with allows (the s) if you do rename

Reply:
Without the s is better imo, also: https://github.com/FabricMC/fabric/blob/c045166c0f58a909dd87d682f92d1230f125638e/fabric-screen-api-v1/src/main/java/net/fabricmc/fabric/api/client/screen/v1/ScreenMouseEvents.java#L48.

Reply:
> 
> 
> Without the s is better imo, also:
> 
> https://github.com/FabricMC/fabric/blob/c045166c0f58a909dd87d682f92d1230f125638e/fabric-screen-api-v1/src/main/java/net/fabricmc/fabric/api/client/screen/v1/ScreenMouseEvents.java#L48
> .

notice the return type.

Reply:
```suggestion
	boolean canUpdateAnimation(ItemStack original, ItemStack updated);
```

Reply:
Ok but the event itself also uses `allow`: https://github.com/FabricMC/fabric/blob/c045166c0f58a909dd87d682f92d1230f125638e/fabric-screen-api-v1/src/main/java/net/fabricmc/fabric/api/client/screen/v1/ScreenMouseEvents.java#L160

Most functions are imperative: `get`, `register`, so why not `allow`? We could also go for `doUpdateAnimation`, but imo `allow` is clearer.

Reply:
allow is like a setter that after you call, you allow something. it is not a getter; allows is a getter.

Reply:
What about the mouse click event then?

Reply:
> 
> 
> What about the mouse click event then?

It's just a bad name. Fabric api has bad names and I am mainly reviewing yarn for consistent names. And back when that pr was merged I wasn't active

Reply:
I disagree (that s is ugly imo) but it's not the end of the world.

Reply:
Imo `can` prefix or `should` prefix (as seen in javadoc) work as well

Reply:
```suggestion
	boolean shouldUpdateAnimation(ItemStack original, ItemStack updated);
```
Consistency with javadoc, so this name. No matter what `boolean updateAnimation` is poor as people think it updates the animation and returns if the update was successful (in case javadoc/parameter names are gone as a result of compiling to jars)

Reply:
`shouldPlayUpdateAnimation` maybe for absolute clarity? 😄

Reply:
Or `shouldAnimateUpdate` :laughing:

Reply:
That's a good one! 😆